src/storage/single_file_block_manager.cpp

```diff
@@ -360,36 +374,44 @@ void SingleFileBlockManager::CreateNewDatabase(QueryContext context) {
 	AddStorageVersionTag();
 
 	MainHeader main_header = ConstructMainHeader(options.version_number.GetIndex());
-	// Derive the encryption key and add it to cache
-	// Unused for plain databases
-	uint8_t salt[MainHeader::SALT_LEN];
-	memset(salt, 0, MainHeader::SALT_LEN);
+
+	// Derive the encryption key and add it to the cache.
+	// Not used for plain databases.
 	data_t derived_key[MainHeader::DEFAULT_ENCRYPTION_KEY_LENGTH];
 
+	// We always need the salt for the unique file identifier.
+	// If encryption is enabled, we generate a random salt.
```

Comment:
This comment is off, I think, currently we always generate and always write, while I think we should generate AND write only if: `encrypted || (storage_version >= 66)` (or maybe simpler, generate only in that case, and write but writes 0 otherwise.